src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -525,18 +526,18 @@ public static XmlSerializer[] FromMappings(XmlMapping[] mappings, Type type)
 
             if ((anySoapMapping && ReflectionMethodEnabled) || Mode == SerializationMode.ReflectionOnly)
             {
-                XmlSerializer[] serializers = GetReflectionBasedSerializers(mappings, type);
+                XmlSerializer[] serializers = GetReflectionBasedSerializers(mappings, type!);
```

Comment:
Is this dereferencing a `null`? `type` here is `null` because in line 512, `null` is being passed in for `type`.

Reply:
you're right, it seems `_rootType` can be null but only if `_mappings` is not null. Will change

Reply:
`_rootType` is non-nullable because the `XmlSerializer`'s ctor doesn't allow type being `null` **but** you can pass `null` through `FromMappings` method and create a bunch of `XmlSerializer` instances with `_rootType` = `null` when `ReflectionMethodEnabled` is true.

That is somewhat inconsistent and looks like a very obscure code path (or maybe not so obscure, I don't know) but seems that either the method should not allow null when condition in line 527 is met or the whole code should adjust to handle `_rootType` as `Type?`.